index.js

```diff
@@ -1,39 +1,99 @@
 async function setupPlugin({config, global}) {
-    global.posthogUrl = config.postHogUrl
+    global.posthogUrl = cleanInstanceUrl(config.postHogUrl)
     global.apiToken = config.postHogApiToken
     global.projectToken = config.postHogProjectToken
-    global.syncScoresIntoPosthog = global.posthogUrl && global.apiToken && global.projectToken
+    global.zapierUrl = config.zapierUrl
+    global.actionIdToName = {}
+    config.actionIdToName.split(',').forEach(pair => {
+        const [actionId, conversionName] = pair.split(':')
+        global.actionIdToName[parseInt(actionId)] = conversionName
+    })
+    global.defaultStartTime = config.defaultStartTime ? new Date(config.defaultStartTime) : new Date(2021, 6, 1)
+    global.isConfigSet = global.posthogUrl
+        && global.apiToken
+        && global.projectToken
+        && global.zapierUrl
+        && Object.keys(global.actionIdToName).length
+        && isValidDate(global.defaultStartTime)
 }
 
+function cleanInstanceUrl(url) {
+    return url.replace(/\/$/, '')
+}
+
+function isValidDate(date) {
+    return date.toString() !== 'Invalid Date'
+}
 
 function addDays(date, days) {
-    var result = new Date(date);
-    result.setDate(result.getDate() + days);
-    return result;
+    const result = new Date(date)
+    result.setDate(result.getDate() + days)
+    return result
+}
+
+function formatTimestampForGoogle(date){
+    const result = new Date(date)
+    if (isValidDate(result)) {
```

Comment:
I have some questions around `isValidDate` - is this a situation where we get a timestamp in a format that for some reason we can't convert to  a JS date (i.e. from bad event data)? Under what situations would date.toString === Invalid Date?

What happens if we end up with an invalid date here and return it to the sheet- is it obviously different and does it break anything?

I wonder if we should be handling this more broadly with a try/catch?

Reply:
In case of receiving an invalid date I'd still rather pass it to the Google sheet than drop the event or stop plugin execution — Google will just skip over that line in the sheet if it has a format it doesn't recognize. We will see the console warning and can manually inspect the sheet if this happens, but I don't expect it to happen.

Reply:
I'm not suggesting we shouldn't send the data to google - what I'm asking is what types of inputs have you seen that lead to an invalid date?

What I want to understand is if the check you have here is safe enough or if this date transform should be wrapped in a try/catch for inputs we might not anticipate?

Reply:
@kpthatsme I think we can anticipate the API response from PostHog to have a valid format no matter what, making any check here just extra cautious. My original concern was that in local testing I had seen dates ending with `Z` and others with `+00:00`. Both are valid ISO formats, but it made me think there is a certain degree of uncertainty here.

Checking if `date === 'Invalid Date'` will prevent an error from being thrown when later calling `toISOString()`, as `new Date('')` and `new Date('foobar')` do not immediately throw errors, but will throw an error if `toISOString()` is called on them.

Reply:
Okay thanks - that's exactly what I was wondering! 

So I think there's a cleaner and more direct way to do this...

In the `formatTimestampForGoogle` function, why not just wrap

```
const result = new Date(date)
       result.setMilliseconds(0)
        return result.toISOString().replace(/\.\d+Z$/, '+0000')
        ```
        
        In a try block and catch it if the error is thrown? It doesn't feel like isValidDate is adding a lot of value aside from handling this error and I think we can get closer to what you're trying to say in the code with this approach?